src/averageAges.js

```diff
@@ -55,10 +82,34 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const childrensWithMoms = people
+    .filter(({ mother }) => mother !== null);
+
+  const menWithMoms = people
+    .filter(({ mother, sex }) => mother !== null && sex === SEX_MALE);
+
+  const childrens = onlyWithSon ? menWithMoms : childrensWithMoms;
```

Comment:
Now everything is written correctly, but if `onlyWithSon` is `false`, then you are doing unnecessary filtering. It is better to first check if this value is `true` and then if it is necessary to do a new filtering.